src/i2c/master.rs

```diff
@@ -619,34 +836,46 @@ impl<M: Mode> embedded_hal_1::i2c::ErrorType for I2cMaster<'_, M> {
     type Error = Error;
 }
 
+impl I2cAddress for u8 {
+    fn to_u16(self) -> u16 {
+        self as u16
+    }
+}
+
+impl I2cAddress for u16 {
+    fn to_u16(self) -> u16 {
+        self
+    }
+}
```

Comment:
I think this is unnecessary. You know the types can only be `u16` or `u8`. In both cases there is a `u16::from()` implementation. Just unconditionally call `addr.into()` and you should be good, no?